nuxeo-ai-core/src/main/java/org/nuxeo/ai/services/DocMetadataServiceImpl.java

```diff
@@ -272,18 +269,100 @@ public DocumentModel resetAuto(DocumentModel doc, String autoField, String xPath
+            List<Map<String, Object>> newSuggestions = suggestions.stream()
+                                                                  .filter(s -> !xPath
+                                                                          .equals(s.get(SUGGESTION_PROPERTY)))
+                                                                  .collect(Collectors.toList());
+            if (!newSuggestions.isEmpty()) {
+                suggestObj.put(SUGGESTION_SUGGESTIONS, newSuggestions);
+                newSuggestList.add(suggestObj);
+            }
+        });
+        doc.setProperty(ENRICHMENT_SCHEMA_NAME, ENRICHMENT_ITEMS, newSuggestList);
+        raiseEvent(doc, ENRICHMENT_MODIFIED, Collections.singleton(xPath), SUGGESTION_SUGGESTIONS);
+        return doc;
+    }
+
+    @Override
+    public DocumentModel removeItemsForDirtyProperties(DocumentModel doc) {
+        List<Map<String, Object>> itemsList = (List) doc.getProperty(ENRICHMENT_SCHEMA_NAME, ENRICHMENT_ITEMS);
+        if (itemsList == null || itemsList.isEmpty()) {
+            return doc;
+        }
+        List<Map<String, Object>> cleanItemsList = new ArrayList<>(itemsList.size());
+        Set<String> removedTargetProperties = new HashSet<>();
+
+        itemsList.forEach(entry -> {
+            String[] props = (String[]) entry.get(ENRICHMENT_INPUT_DOCPROP_PROPERTY);
+            Set<String> inputProperties = props == null ? Collections.emptySet() : new HashSet<>(Arrays.asList(props));
+            if (hadBeenModified(doc, inputProperties)) {
+                List<Map<String, Object>> suggestions = (List<Map<String, Object>>) entry.get(SUGGESTION_SUGGESTIONS);
+                Set<String> targetProps = suggestions.stream()
+                                                     .map(s -> (String) s.get(SUGGESTION_PROPERTY))
+                                                     .collect(Collectors.toSet());
+                removedTargetProperties.addAll(targetProps);
+            } else {
+                cleanItemsList.add(entry);
+            }
+        });
+        if (cleanItemsList.size() != itemsList.size()) {
+            //We made some changes lets update
+            doc.setProperty(ENRICHMENT_SCHEMA_NAME, ENRICHMENT_ITEMS, cleanItemsList);
+            raiseEvent(doc, ENRICHMENT_MODIFIED, removedTargetProperties, "Dirty inputs");
+        }
+        return doc;
+    }
+
+    /**
+     * Have one of the supplied properties been modified?
+     */
+    public static boolean hadBeenModified(DocumentModel doc, Set<String> props) {
+
+        if (props != null) {
+            for (String propName : props) {
+                try {
+                    Property prop = doc.getProperty(propName);
+                    if (prop != null && prop.isDirty()) {
+                        return true;
+                    }
+                } catch (PropertyNotFoundException e) {
```

Comment:
it's a runtime exception. is there a way to avoid that

Reply:
Good question, its standard Nuxeo `PropertyNotFoundException`.  At some point we need to handle this. I am open to suggestions.

Reply:
idk. it's fine since the intention is very clear. Exceptions are not designed good in Nuxeo. Their usage contradicts with common practices.